tests/testHDF5IO.cpp

```diff
@@ -274,4 +264,35 @@ TEST_CASE("writeAttributes", "[hdf5io]")
 
   // close file
   hdf5io.close();
+}
+
+TEST_CASE("useSWMRmode", "[hdf5io]")
+{
+  // create and open file
+  std::string path = getTestFilePath("test_swmr.h5");
+  std::unique_ptr<HDF5::HDF5IO> hdf5io = std::make_unique<HDF5::HDF5IO>(path);
+  hdf5io->open();
+
+  // add a dataset
+  std::vector<int> testData = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+  std::string dataPath = "/data";
+  SizeType numSamples = testData.size();
+  std::unique_ptr<BaseRecordingData> dataset = hdf5io->createArrayDataSet(
+      BaseDataType::I32, SizeArray {0}, SizeArray {1}, dataPath);
+
+  // turn on swmr mode
+  Status status = hdf5io->startRecording();
+  REQUIRE(status == Status::Success);
+
+  // write data block and flush
+  std::vector<SizeType> dataShape = {numSamples};
+  std::vector<SizeType> positionOffset = {0};
+  dataset->writeDataBlock(
+      dataShape, positionOffset, BaseDataType::I32, &testData[0]);
+  H5Dflush(static_cast<HDF5::HDF5RecordingData*>(dataset.get())
+               ->getDataSet()
+               ->getId());
```

Comment:
Why is the `H5Dflush` needed here? Shouldn't `hdf5io->close()` take care of flushing the data to disk?

Reply:
Yes as you mentioned below, I was using H5Dflush so that I could manually check that the data could be read by a different process before closing the file.

I've been trying to setup the tests to check that reading the data would fail if the file was not in SWMR mode. However, this seems to be more complicated than I expected, as it turns out that reading when the file is not in SWMR mode will work fine as long as it's in the same process, even if I set up multiple threads (see [related discussion here](https://forum.hdfgroup.org/t/swmr-r-w-in-two-processes-good-in-same-process-bad/10464/6)).

I was trying to setup multiple processes with `fork`, but it turns out Catch2 has some [known limitations](https://github.com/catchorg/Catch2/blob/devel/docs/limitations.md) related to this. You suggested a third option of calling a command line script to read from the file above, so I can try doing that instead.

Reply:
> You suggested a third option of calling a command line script to read from the file above, so I can try doing that instead

Since this is for improving testing, I think it would be find to address this in a separate issue, i.e., we shouldn't block merging this PR because of this particular issue since we know it is working.

Reply:
I think I got it working!